packages/beacon-node/src/chain/prepareNextSlot.ts

```diff
@@ -240,4 +244,26 @@ export class PrepareNextSlotScheduler {
     state.hashTreeRoot();
     hashTreeRootTimer?.();
   }
+
+  /**
+   * Stop eth1 data polling after eth1DepositIndex has reached depositRequestsStartIndex in Electra as described in EIP-6110
+   */
+  stopEth1Polling(): void {
+    // Only continue if eth1 is still polling. State regen is expensive
+    if (this.chain.eth1.isPollingEth1Data()) {
+      const finalizedCheckpoint = this.chain.forkChoice.getFinalizedCheckpoint();
+      const fork = this.config.getForkInfoAtEpoch(finalizedCheckpoint.epoch).name;
+
+      if (isForkPostElectra(fork)) {
+        const finalizedState = this.chain.getStateByCheckpoint(finalizedCheckpoint)?.state;
+        if (
+          finalizedState !== undefined &&
+          finalizedState.eth1DepositIndex >= (finalizedState as BeaconStateElectra).depositRequestsStartIndex
```

Comment:
not that it matters much but can this just be a strict equal?

```suggestion
          finalizedState.eth1DepositIndex === (finalizedState as BeaconStateElectra).depositRequestsStartIndex
```